File: js/count-primes.js
function countPrimes(n) {
  let primeNumbers = Array(n);
  for (let i = 0; i < n; i++) {
    primeNumbers[i] = true;
  }
  primeNumbers[0] = false;
  primeNumbers[1] = false;
  let last = parseInt(Math.sqrt(n));
  for (let i = 2; i < last; i++) {
    if (!primeNumbers[i]) {
      continue;
    }
    for (let j = i * 2; j < n; j += i) {
      primeNumbers[j] = false;
    }
  }
  let c = 0;
  for (let i = 0; i < n; i++) {
    if (primeNumbers[i]) {
      c += 1;
    }
  }
  return c;
}

function main() {
  let durations = Array(100);
  for (let i = 0; i < durations.length; i++) {
    let start = new Date().getTime();
    let n = countPrimes(1000000);
    durations[i] = (new Date().getTime()) - start;
    if (n != 78498) {
      console.log(`found ${n} instead of 78498 prime numbers`);
      return -1;
    }
    console.log(`Iteration ${i} completed in ${durations[i]}ms`);
  }

  let min = durations[0];
  let max = durations[0];
  let total = durations[0];
  for (let i = 1; i < durations.length; i++) {
    if (durations[i] < min) {
      min = durations[i];
    }

    if (durations[i] > max) {
      max = durations[i];
    }

    total += durations[i];
  }
  console.log(`${durations.length} iterations completed in ${total/1000.0}s, min: ${min}ms ` + 
    `max: ${max}ms average: ${total / durations.length}ms`);
}

main();

File: python/count-primes.py
import math
import sys
import time

def count_primes(n):
    prime_numbers = [True for i in range(n)]
    prime_numbers[0] = False
    prime_numbers[1] = False

    last = int(math.sqrt(n))

    for i in range(2, last):
        if not prime_numbers[i]:
            continue
        j = i * 2
        while j < n:
            prime_numbers[j] = False
            j += i

    c = 0
    for i in prime_numbers:
        if i:
            c += 1

    return c


if __name__ == '__main__':
    durations = [0.0 for i in range(100)]
    for i in range(len(durations)):
        start = time.time()
        n = count_primes(1000000)
        durations[i] = (time.time() - start) * 1000
        if n != 78498:
            print(f"found {n} instead of 78498 prime numbers")
            sys.exit(-1)
        print(f"Iteration {i} completed in {durations[i]}ms")

    min_duration = durations[0]
    max_duration = durations[0]
    total = durations[0]
    for i in range(1, len(durations)):
        if durations[i] < min_duration:
            min_duration = durations[i]

        if durations[i] > max_duration:
            max_duration = durations[i]

        total += durations[i]
    avg = total / len(durations)

    print(f"{len(durations)} iterations completed in {total / 1000}s, min: {min_duration}ms max: {max_duration}ms average: {avg}ms")

